hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/CommonConsumerProperties.java

```diff
@@ -0,0 +1,168 @@
+        return healthCheckPort;
+    }
+
+    public void setHealthCheckPort(int healthCheckPort) {
+        this.healthCheckPort = healthCheckPort;
+    }
+
+    public boolean isFilteringEnabled() {
+        return filteringEnabled;
+    }
+
+    public void setFilteringEnabled(boolean filteringEnabled) {
+        this.filteringEnabled = filteringEnabled;
+    }
+
+    public long getSubscriptionIdsCacheRemovedExpireAfterAccessSeconds() {
+        return subscriptionIdsCacheRemovedExpireAfterAccessSeconds;
+    }
+
+    public void setSubscriptionIdsCacheRemovedExpireAfterAccessSeconds(long subscriptionIdsCacheRemovedExpireAfterAccessSeconds) {
+        this.subscriptionIdsCacheRemovedExpireAfterAccessSeconds = subscriptionIdsCacheRemovedExpireAfterAccessSeconds;
+    }
+
+    public BackgroundSupervisor getBackgroundSupervisor() {
+        return backgroundSupervisor;
+    }
+
+    public void setBackgroundSupervisor(BackgroundSupervisor backgroundSupervisor) {
+        this.backgroundSupervisor = backgroundSupervisor;
+    }
+
+    public int getSignalProcessingIntervalMilliseconds() {
+        return signalProcessingIntervalMilliseconds;
+    }
+
+    public void setSignalProcessingIntervalMilliseconds(int signalProcessingIntervalMilliseconds) {
+        this.signalProcessingIntervalMilliseconds = signalProcessingIntervalMilliseconds;
+    }
+
+    public int getSignalProcessingQueueSize() {
+        return signalProcessingQueueSize;
+    }
+
+    public void setSignalProcessingQueueSize(int signalProcessingQueueSize) {
+        this.signalProcessingQueueSize = signalProcessingQueueSize;
+    }
+
+    public boolean isUseTopicMessageSizeEnabled() {
+        return useTopicMessageSizeEnabled;
+    }
+
+    public void setUseTopicMessageSizeEnabled(boolean useTopicMessageSizeEnabled) {
+        this.useTopicMessageSizeEnabled = useTopicMessageSizeEnabled;
+    }
+
+    public String getClientId() {
+        return clientId;
+    }
+
+    public void setClientId(String clientId) {
+        this.clientId = clientId;
+    }
+
+    public static final class BackgroundSupervisor {
+
+        private int interval = 20_000;
+
+        private int unhealthyAfter = 600_000;
+
+        private int killAfter = 300_000;
+
+        public int getInterval() {
+            return interval;
+        }
+
+        public void setInterval(int interval) {
+            this.interval = interval;
+        }
+
+        public int getUnhealthyAfter() {
+            return unhealthyAfter;
+        }
+
+        public void setUnhealthyAfter(int unhealthyAfter) {
+            this.unhealthyAfter = unhealthyAfter;
+        }
+
+        public int getKillAfter() {
+            return killAfter;
+        }
+
+        public void setKillAfter(int killAfter) {
+            this.killAfter = killAfter;
+        }
+    }
+
+    public CommonConsumerParameters toCommonConsumerParameters() {
+        return new CommonConsumerParameters(
+                this.threadPoolSize,
```

Comment:
I know the comment I'm giving would cause big change but I think about such a constructor. It is an antipattern to have calls with so many parameters. It is very easy to call this constructor with exchanged parameters (the ones with the same type next to each other). Maybe we should add the another task to use builders in such a situations ?